extensions/cornerstone/src/services/ColorbarService/ColorbarService.ts

```diff
@@ -78,35 +70,27 @@ export default class ColorbarService extends PubSubService {
       return;
     }
 
-    const { element } = viewport;
     const actorEntries = viewport.getActors();
-
     if (!actorEntries || actorEntries.length === 0) {
       return;
     }
 
-    const { position, width: thickness, activeColormapName, colormaps } = options;
-
-    const numContainers = displaySetInstanceUIDs.length;
+    const { activeColormapName, colormaps } = options;
 
-    const containers = this.createContainers(
-      numContainers,
-      element,
-      position,
-      thickness,
-      viewportId
-    );
-
-    displaySetInstanceUIDs.forEach((displaySetInstanceUID, index) => {
+    displaySetInstanceUIDs.forEach(displaySetInstanceUID => {
       // don't show colorbar for overlay display sets (e.g. segmentation)
       const displaySet = displaySetService.getDisplaySetByUID(displaySetInstanceUID);
       if (displaySet.isOverlayDisplaySet) {
         return;
       }
 
       const volumeId = this.getVolumeIdForIdentifier(viewport, displaySetInstanceUID);
-      const properties = viewport?.getProperties(volumeId);
+      const properties =
+        viewport instanceof VolumeViewport
+          ? viewport.getProperties(volumeId)
```

Comment:
Can you just pass in a dataId that identifies the specific layer all the time?  I really dislike this viewport instanceof VolumeViewport handling of query parameters.  Rather just make it generic and let the internals decide if it uses the data/volumeId